src/servers/trigger.ts

```diff
@@ -181,7 +181,7 @@ export const onchainBlockEventHandler = async (height: number) => {
 export const invoiceUpdateEventHandler = async (invoice: GetInvoiceResult) => {
   logger.info({ invoice }, "invoiceUpdateEventHandler")
 
-  if (!invoice.is_confirmed) {
+  if (!invoice.is_held) {
```

Comment:
maybe this if branch as an early return is unnecessary/a duplicate?

we are doing it also in https://github.com/GaloyMoney/galoy/pull/1395/files#diff-f02445547836eff187be3a4b54e41d92b4223dd7c304847b688ef3016eec317fR108

Reply:
I mean un unnecessary optimization that make the flow more complex

Reply:
You mean we should remove these 2 lines from the trigger?

If we do, it would mean unnecessarily doing 2 external calls until we get to that line again:
- invoices collection mongo lookup
- lnd invoice lookup

Reply:
If we decide to do this, it may be better as a separate PR so we can isolate any changed effects on this flow